### PR TITLE
[Al Amin Step-11] Let's set the validation

### PR DESCRIPTION
**Pull request by rr-mdalauddin-alamin**
### Step 11: Let's set the validation

- Let's set the validation
  - Think about which validation to add to which column
  - Let's create a migration that also sets DB constraints
  - Create with a `rails generate` command to create only the migration file
- Let's display a validation error message on the screen
- Let's write a model test for validation
- Create a PR on GitHub for review

### CODE REVIEW

File: myapp/app/models/user.rb
Comment (ksak19) [inline myapp/app/models/user.rb:1]: naruhodo. You already applied validation in the previous steps 👍

File: myapp/spec/models/task_spec.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+    context 'when valid' do
+      it 'is valid with valid attributes' do
+        task = build(:task, user: user)
+        expect(task).to be_valid
+      end
+    end
+
+    context 'when invalid' do
+      shared_examples 'invalid task' do |attribute|
```
Comment (ksak19) [inline myapp/spec/models/task_spec.rb:15]: `shared_examples` 👍 
It's too much if there is only one single test case but you might add more in the following steps, maybe?

**Review by ksak19 — APPROVED** [review]
lgtm

File: myapp/app/models/user.rb
Comment (rr-mdalauddin-alamin) [reply myapp/app/models/user.rb:1]: Yep, there are many hidden features i already done it before. it was a misunderstanding.

File: myapp/spec/models/task_spec.rb (hunk 1 above)
Comment (rr-mdalauddin-alamin) [reply myapp/spec/models/task_spec.rb:15]: umm.. I'll think about it. thanks for the gesture!

**Pull request merged**